### DISCUSSION TITLE
[Q] Is it possible to use this library on Raspberry Pi 3?

### DISCUSSION THREAD

**Discussion by don-boucher** [googleapis/google-cloud-cpp#6807] [Q&A] [answered]
Just as the title says, I'm trying to use this library on the raspberry pi 3, but I'm running into an error.

```
~/google-cloud-cpp $ cmake --build cmake-out -- -j $(nproc)
[  1%] Built target ssl-project                                                                                          c-project-build-*.log
[ 16%] Built target abseil-cpp-project
[ 16%] Built target nlohmann-json-project
[ 24%] Built target crc32c-project
[ 41%] Built target re2-project
[ 41%] Built target zlib-project
[ 49%] Built target c-ares-project
[ 57%] Built target benchmark-project
[ 74%] Built target curl-project
[ 74%] Built target googletest-project
[ 82%] Built target protobuf-project
[ 83%] Performing build step for 'grpc-project'
CMake Error at /home/pi/google-cloud-cpp/cmake-out/external/grpc/src/grpc-project-stamp/grpc-project-build-.cmake:49 (mes

sage):
  Command failed: 2                                                                                                      sage):

   '/usr/bin/cmake' '--build' '/home/pi/google-cloud-cpp/cmake-out/external/grpc/src/grpc-project-build' '--' '-j' '4'

  See also

    /home/pi/google-cloud-cpp/cmake-out/external/grpc/src/grpc-project-stamp/grpc-project-build-*.log


make[2]: *** [CMakeFiles/grpc-project.dir/build.make:117: external/grpc/src/grpc-project-stamp/grpc-project-build] Error
1                                                                                                                        1
make[1]: *** [CMakeFiles/Makefile2:240: CMakeFiles/grpc-project.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

Can this library be used for raspberry pi 3?

**Comment by coryan** [comment, marked as answer]
**TL;DR;** we don't know, we have never tested on that platform.

Longer answer: if the dependencies build, then the library probably will build too.  Candidly, it is unlikely that we will be able to prioritize the work to compile on a raspberry pi (any model), but if you send us PRs to fix problems we would like to merge them.  If you send us questions we would be happy to help.

It seems that you are using a super build, while that is convenient, it is not the best way to troubleshoot problems *and* it is not designed to support cross-compilation (which you may nor may not be using).  In this case the error says [gRPC](https://github.com/grpc/grpc) did not compile for you, it is hard to see why from the logs you sent.

I would start by manually building the dependencies (Abseil, CRC32C, Protobuf, gRPC, etc. -- see the doc/packaging.md document for details), then try to compile the library against those dependencies. Alternatively, consider using `vcpkg` which seems to support raspberry pi 3 (see microsoft/vcpkg#9570) **and** includes a port for `google-cloud-cpp`.

**Comment by don-boucher** [comment]
Thanks for the quick response!

Ultimately, I want to get access to datastore using the client credentials oauth flow.
I'm familiar with the python and golang packages for gcp, so I might end up using one of those. I was interested in the c++ library because the existing project that I'm working on is all in c++.

I'll take a look at your two suggestions.

Thanks!
